### PR TITLE
chore(package.json, package-lock.json, gulpfile.js): removes gulp in favor of babel-cli for build an

### PR DESCRIPTION
**Pull request by drewbrokke**

### CODE REVIEW

**Comment by drewbrokke** [comment]
Third time's a charm, I hope.

@jwu910 this pull does not add any new features or meaningful changes to the behavior. It is intended to address a few miscellaneous items:

- [ ] Use modern import statements where possible. This allows editors like VSCode to pull in type information and analyze the code more accurately.
- [ ] Use async/await syntax for Promises. IMO the code is easier to reason about this way.
- [ ] Update the doc blocks in certain spots to accurately reflect what is going to be returned. A couple were out of date
- [ ] Remove `gulp` from the build chain since it was basically just calling `babel` anyways, and `babel-cli` has a watch mode with more thorough output.

Please let me know if you have any questions or comments.  Thanks!

**Comment by jwu910** [comment]
Just started reviewing :)

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "prepublishOnly": "npm run build",
     "start": "node index.js",
     "test": "npm run lint",
-    "watch": "gulp watch"
+    "watch": "npx babel src --extensions .js --out-dir dist --verbose --watch"
```
Comment (jwu910) [inline package.json]: Nice catch on the unnecessary gulp dependency, but we also would not need `npx` in the case. NPM should be able to use the local babel dependency to run this.

Comment (drewbrokke) [reply package.json]: 👍 just force-pushed removing `npx`

**Comment by jwu910** [comment]
Reviewed, Thanks!

**Pull request merged**